addons/bus/tests/cleanup.py

```diff
@@ -0,0 +1,40 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from unittest.mock import patch
+from odoo.tests import HttpCase
+from ..websocket import Websocket, ConnectionState, CloseCode
+
+
+def cleanup_websockets_after_test():
+    """
+    Close all the websocket instances that were opened during the test. This
+    method should be called after each test to ensure no websocket is left open.
+    """
+    for websocket in Websocket._instances:
+        if websocket.in_test_mode and websocket.state is ConnectionState.OPEN:
+            # The websocket linked to the browser was not closed after the
+            # test. Close the connection without waiting for a response.
+            websocket.disconnect(CloseCode.ABNORMAL_CLOSURE)
+
+
+def register_websocket_cleanup(http_case):
+    """
+    Patch the websocket `__init__` method to mark it as being in test mode.
+    This information is then used to ensure every websocket is closed at the
+    end of the test.
+    """
+    original_init = Websocket.__init__
+
+    def mocked_init(self, *a, **kw):
+        original_init(self, *a, **kw)
+        self.in_test_mode = True
+
+    http_case._websocket_init_patch = patch.object(
+        Websocket, "__init__", autospec=True, side_effect=mocked_init
+    )
+    http_case.startPatcher(http_case._websocket_init_patch)
+    http_case.addCleanup(http_case._websocket_init_patch.stop)
+    http_case.addCleanup(cleanup_websockets_after_test)
+
+
+HttpCase.test_setups.append(register_websocket_cleanup)
```

Comment:
I fear that this may not work in some corner case, like updating a module depending on bus but without updating buf. Something like -u crm --test-enable. This is the old way to run tests but still kind of supported. I this case I think that this won't be executed (to check)